server/src/main/java/org/apache/druid/server/metrics/ServiceStatusMonitor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.metrics;
+
+import com.google.common.base.Supplier;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.druid.java.util.metrics.AbstractMonitor;
+
+import java.util.Map;
+
+/**
+ * Monitor service running status.
+ * For Overlord/Coordinator, the metric reported is service leader count.
+ */
+public class ServiceStatusMonitor extends AbstractMonitor {
+
+  @Named("heartbeat")
+  @Inject(optional = true)
+  Supplier<Map<String, Number>> heartbeatTagsSupplier = null;
+
+  @Inject
+  public ServiceStatusMonitor() {
+  }
+
+  @Override
+  public boolean doMonitor(ServiceEmitter emitter) {
+    if (heartbeatTagsSupplier == null) {
+      return true;
+    }
+
+    heartbeatTagsSupplier.get().forEach((k, v) -> {
```

Comment:
Please add a null check on the supplier, just in case someone wants to use this monitor for other services, where the tags supplier is not being injected.

Reply:
Updated.

Reply:
To make the heartbeat work on other services, I think the pattern in this monitor should be

```
ServiceMetricEventBuilder builder  = ...;
if (heartbeatDimensions is not empty) {
  heartbeatDimensions.forEach(builder.setDimension(k, v);
}
emitter.emit(builder.build("druid/heartbeat", 1);
```